src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepAtomNode.java

```diff
@@ -48,7 +48,7 @@
 
     public StepAtomNode(CpsFlowExecution exec, StepDescriptor d, FlowNode parent) {
         super(exec, exec.iotaStr(), parent);
-        this.descriptorId = d!=null ? d.getId() : null;
+        this.descriptorId = d!=null ? d.getId().intern() : null;
```

Comment:
Gratuitous. `Descriptor.getId()` normally calls `Class.getName()`, which uses a cached value; and `Descriptor`s are singletons, so you are not going to have multiple copied of a given `id` to begin with.